homeassistant/components/izone/climate.py

```diff
@@ -0,0 +1,511 @@
+    def target_temperature(self) -> Optional[float]:
+        """Return the temperature we try to reach."""
+        if not self._supported_features & SUPPORT_TARGET_TEMPERATURE:
+            return None
+        return self._controller.temp_setpoint
+
+    @property
+    def supply_temperature(self) -> float:
+        """Return the current supply, or in duct, temperature."""
+        return self._controller.temp_supply
+
+    @property
+    def target_temperature_step(self) -> Optional[float]:
+        """Return the supported step of target temperature."""
+        return 0.5
+
+    @property
+    def fan_mode(self) -> Optional[str]:
+        """Return the fan setting."""
+        return _IZONE_FAN_TO_HA[self._controller.fan]
+
+    @property
+    def fan_modes(self) -> Optional[List[str]]:
+        """Return the list of available fan modes."""
+        return list(self._fan_to_pizone)
+
+    @property
+    def min_temp(self) -> float:
+        """Return the minimum temperature."""
+        return self._controller.temp_min
+
+    @property
+    def max_temp(self) -> float:
+        """Return the maximum temperature."""
+        return self._controller.temp_max
+
+    async def wrap_and_catch(self, coro):
+        """Catch any connection errors and set unavailable."""
+        try:
+            await coro
+        except ConnectionError as ex:
+            self.set_available(False, ex)
+        else:
+            self.set_available(True)
+
+    async def async_set_temperature(self, **kwargs) -> None:
+        """Set new target temperature."""
+        if not self.supported_features & SUPPORT_TARGET_TEMPERATURE:
+            self.async_schedule_update_ha_state(True)
+            return
+        temp = kwargs.get(ATTR_TEMPERATURE)
+        if temp is not None:
+            await self.wrap_and_catch(
+                self._controller.set_temp_setpoint(temp))
+
+    async def async_set_fan_mode(self, fan_mode: str) -> None:
+        """Set new target fan mode."""
+        fan = self._fan_to_pizone[fan_mode]
+        await self.wrap_and_catch(self._controller.set_fan(fan))
+
+    async def async_set_hvac_mode(self, hvac_mode: str) -> None:
+        """Set new target operation mode."""
+        if hvac_mode == HVAC_MODE_OFF:
+            await self.wrap_and_catch(self._controller.set_on(False))
+            return
+        if not self._controller.is_on:
+            await self.wrap_and_catch(self._controller.set_on(True))
+        if self._controller.free_air:
+            return
+        mode = self._state_to_pizone[hvac_mode]
+        await self.wrap_and_catch(self._controller.set_mode(mode))
+
+    async def async_set_preset_mode(self, preset_mode: str) -> None:
+        """Set the preset mode."""
+        await self.wrap_and_catch(
+            self._controller.set_free_air(preset_mode == PRESET_ECO))
+
+    async def async_turn_on(self) -> None:
+        """Turn the entity on."""
+        await self.wrap_and_catch(self._controller.set_on(True))
+
```

Comment:
We can't turn the controller entity off?

Reply:
This is now handled in the climate 1.0 way, the superclass calls `async_set_hvac_mode(HVAC_MODE_OFF)` which then turns it off.